src/ray/core_worker/core_worker.cc

```diff
@@ -1565,22 +1572,23 @@ Status CoreWorker::Get(const std::vector<ObjectID> &ids,
 
   // ray.get path for experimental.Channel objects.
   if (is_experimental_channel) {
-    if (timeout_ms >= 0) {
-      return Status::NotImplemented(
-          "non-infinity timeout_ms not supported for experimental channels");
-    }
-    return GetExperimentalMutableObjects(ids, results);
+    int64_t expiration_ms = timeout_ms < 0 ? -1 : current_time_ms() + timeout_ms;
```

Comment:
nit: make the definition of `timeout_ms` consistent with `ExperimentalChannelWriteAcquire`. 

```
int64_t expiration_ms = timeout_ms == -1 ? -1 : current_time_ms() + timeout_ms;
```

Reply:
Fixed after refactoring.